code/controllers/subsystems/processing/odyssey.dm

```diff
@@ -0,0 +1,225 @@
+SUBSYSTEM_DEF(odyssey)
+	name = "Odyssey"
+	init_order = INIT_ORDER_ODYSSEY
+	flags = SS_BACKGROUND
+	wait = 8
+	can_fire = FALSE
+
+	/// The selected scenario singleton.
+	var/singleton/scenario/scenario
+	/// The z-levels that the odyssey takes place in. If null, it means that there are no loaded zlevels for this odyssey.
+	/// It probably takes place on the Horizon in that case.
+	var/scenario_zlevel
+	/// This is the site the odyssey takes place on. If null, then the mission takes place on a non-site zlevel.
+	/// Should only be changed through set_scenario_site().
+	var/datum/map_template/ruin/away_site/scenario_site
+	/// A list of currently spawned actors for easy access.
+	var/list/actors
+	/// A list of currently spawned storytellers for easy access.
+	var/list/storytellers
+	/// Whether or not we've sent the odyssey's roundstart report yet.
+	var/has_sent_roundstart_announcement = FALSE
+	/// The Horizon's overmap object. We keep it here for convenience.
+	var/obj/effect/overmap/visitable/ship/horizon
+
+/datum/controller/subsystem/odyssey/Initialize()
+	return SS_INIT_SUCCESS
+
+/datum/controller/subsystem/odyssey/Recover()
+	scenario = SSodyssey.scenario
+	scenario_site = SSodyssey.scenario_site
+	scenario_zlevel = SSodyssey.scenario_zlevel
+	actors = SSodyssey.actors
+	storytellers = SSodyssey.storytellers
+
+/datum/controller/subsystem/odyssey/fire()
+	. = ..()
+	if(!has_sent_roundstart_announcement)
+		addtimer(CALLBACK(scenario, TYPE_PROC_REF(/singleton/scenario, notify_horizon_early), horizon), rand(4 MINUTES, 6 MINUTES))
+		addtimer(CALLBACK(scenario, TYPE_PROC_REF(/singleton/scenario, notify_horizon_late), horizon), rand(20 MINUTES, 30 MINUTES))
+		has_sent_roundstart_announcement = TRUE
+
+/**
+ * Picks a random odyssey while keeping in mind sector requirements.
+ * If successful, makes the SS start firing.
+ */
+/datum/controller/subsystem/odyssey/proc/pick_odyssey()
+	var/list/all_scenarios = GET_SINGLETON_SUBTYPE_LIST(/singleton/scenario)
+	var/list/possible_scenarios = list()
+	for(var/singleton/scenario/S in all_scenarios)
+		if((SSatlas.current_sector.name in S.sector_whitelist) || !length(S.sector_whitelist))
+			possible_scenarios[S] = S.weight
+
+	if(!length(possible_scenarios))
+		log_debug("CRITICAL ERROR: No available odyssey for sector [SSatlas.current_sector.name]!")
```

Comment:
From the function header:
```
/**
 * Use the macro LOG_DEBUG instead of calling this directly.
 * Used for general_purpose debug logging, please use more specific ones if available, or consider adding your own
 */
```

This should probably need to use either the gamemode logging facility, or have its own logging facility implemented, like other subsystems